File: Ejercicio4/pwLengthFilter.rb
require_relative 'filter'

class PwLengthFilter
  include Filter

  def ejecutar(credenciales)
    if credenciales.password.length >= 10
      puts "La contraseña contiene un número de caracteres adecuado."
      return true
    else
      puts "Su contraseña es muy débil, por favor, añadele 
      al menos 10 caracteres."
      return false
    end
  end
end

File: P2/Ejercicio1/lib/pwLengthFilter.dart
import 'filter.dart';
import 'struct_credenciales.dart';

class PwLengthFilter implements Filter {
  static const int _minLength = 10;
  static const int _maxLength = 128;
  String _error = "";

  @override
  bool ejecutar(Credenciales mensaje) {

    if (mensaje.password.isEmpty) {
      _error = "La contraseña no puede estar vacía";
      return false;
    }
    
    if (mensaje.password.length < _minLength) {
      _error = 'La contraseña debe tener al menos $_minLength caracteres';
      return false;
    }
    
    if (mensaje.password.length > _maxLength) {
      _error = "La contraseña no puede exceder $_maxLength caracteres";
      return false;
    }

    return true;
  }

  @override
  String msgerror(){
    return _error;
  }
}